python-package/lightgbm/plotting.py

```diff
@@ -1,13 +1,14 @@
 # coding: utf-8
 """Plotting library."""
+import math
 from copy import deepcopy
 from io import BytesIO
-from typing import Any, Dict, List, Optional, Tuple, Union
+from typing import Any, Dict, List, Literal, Optional, Tuple, Union
```

Comment:
Nice improvement compared to general `: str` typing!
But unfortunately, `Literal` is only available in Python 3.8 and newer while LightGBM should support Python 3.7 as well.
> New in version 3.8.
   https://docs.python.org/3/library/typing.html#typing.Literal

https://github.com/microsoft/LightGBM/blob/560c884dd153ac6a95429df8c749aae9632b8215/python-package/setup.py#L386-L390

I believe we should update `: str` type hints with `Literal` in many places in the codebase when Python 3.7 will reach its EOL ([`2023-06-27`](https://devguide.python.org/#status-of-python-branches)).

Reply:
We could maybe consider using [typing-extensions](https://pypi.org/project/typing-extensions/)

Reply:
I'm personally against introducing new package dependencies just for typehints.